modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -297,6 +298,7 @@ object Inputs {
       else if os.isDir(path) then Right(Seq(Directory(path)))
       else if acceptFds && arg.startsWith("/dev/fd/") then
         Right(Seq(VirtualScript(content, arg, os.sub / s"input-${idx + 1}.sc")))
+      else if isRunWithShebang then Right(Seq(Script(dir, subPath)))
```

Comment:
Check if file has no extension.

Reply:
technically if it has some extension that isn't supported by Scala CLI, but it does have the `shebang` header, I think we should be able to run it...